stdlib/public/runtime/Casting.cpp

```diff
@@ -349,42 +346,38 @@ TwoWordPair<const char *, uintptr_t>::Return
 swift_getTypeName(const Metadata *type, bool qualified) {
   using Pair = TwoWordPair<const char *, uintptr_t>;
   using Key = llvm::PointerIntPair<const Metadata *, 1, bool>;
-  
-  static pthread_rwlock_t TypeNameCacheLock = PTHREAD_RWLOCK_INITIALIZER;
+
+  static ReadWriteLock TypeNameCacheLock;
```

Comment:
I need a little guidance on code that is using a static Mutex or ReadWriteLock. Such code currently gets warning like the following. 
```
[8/126] Building CXX object stdlib/public/runtime/CMakeFiles/swiftRuntime-macosx-x86_64.dir/Casting.cpp.o
/Users/shawnce/Development/apple_swift/swift/stdlib/public/runtime/Casting.cpp:350:24: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  static ReadWriteLock TypeNameCacheLock;
```
So should I attempt to avoid static storage for locks, provide a variant that can be statically stored, suppress the warning, or? Note – while unlikely – non-pthread implementation may require or strong desire deallocation to clean up the lock implementations.